### DISCUSSION TITLE
Can i use Dynamic Expression<Func<,>> as MinimalApi Delegate?

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [dotnet/aspnetcore#56410] [Q&A]
how can i generate delegate for minimalAPI delegates in asp.net core?

**Comment by davidfowl** [comment]
It’s unclear what you want to do but if you compile the expression it should work.

**Comment by sajjadarashhh** [comment]
@davidfowl when we Compile LambdaExpressions that have another parameter that isnt show us and that have'nt name and type of that is : 'Closure'
and that is'nt have name

**Reply by davidfowl** [reply]
Can you write a minimal example of what you are trying to do and shows what doesn’t work? It might be some detail of the generated delegate that makes it not work well.

**Reply by davidfowl** [reply]
Ah I think you’re saying that the parameter info for generated delegates don’t have a name. Yep that’s a problem for minimal APIs. There’s also no way to apply attributes to the parameters using expression tree APIs. 

You might need to open up an issue on dotnet/runtime with this scenario.

**Reply by sajjadarashhh** [reply]
Tats not very hard to try it just you need to this for your delegates
((Expression<Func<int,int>>)Input=>input)).Compile()
Thats not working to set for delegate argument 
@davidfowl

**Reply by davidfowl** [reply]
Can you share a real example of what you're trying to do?

**Reply by nquandt** [reply]
I'm not sure if this is in the realm of what you are trying to do, but I have some code in which I am generating delegates and methods at runtime with Reflection.Emit.

https://github.com/nquandt/femur/blob/master/src/AspNetCore/Endpoints/src/DelegateGenerator.cs

and here is a link to a discussion I have discussing why/where I'm using this functionality.
https://github.com/dotnet/aspnetcore/discussions/60194
